### PR TITLE
ci(add-code-coverage-ci-script): script to upload the code coverage

### LINKED ISSUE #23: Add the missing badges in the project README
There are well-known badges for the projects such as maintainability, and test passing percentage. It will enhance the project's credibility

### PR DESCRIPTION
**Pull request by hamzashoukat94**
Code climate tells you the key metrics of your code. Added the CI script to upload the coverage on code climate which will later be used for badges.

closes #23

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+name: Update the Code Coverage
+
+on:
+  workflow_dispatch
```
Comment (sufyankhanrao) [inline .github/workflows/test.yml:4]: This workflow should be triggered upon pushing to the main branch only and should not run whenever the release workflow pushes that release version update in the POM file if you have that step of pushing POM file change in the release script.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+name: Update the Code Coverage
```
Comment (sufyankhanrao) [inline .github/workflows/test.yml:1]: This workflow should be renamed to `Publish Test Coverage Report`.

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+on:
+  workflow_dispatch
+
+jobs:
+  coverage:
```
Comment (sufyankhanrao) [inline .github/workflows/test.yml:7]: The job should be renamed to publish-code-coverage-report.

**Review by sufyankhanrao — CHANGES_REQUESTED** [review]
Above mentioned changes need your attention.

File: .github/workflows/test.yml (hunk 1 above)
Comment (hamzashoukat94) [reply .github/workflows/test.yml]: If the main branch's last commit message contains `ci skip` word then this workflow will not run. And In the release script, I will add this `ci skip` keyword in the commit message of POM changes.

File: .github/workflows/test.yml (hunk 2 above)
Comment (hamzashoukat94) [reply .github/workflows/test.yml]: Renamed to `Publish Test Coverage Report`

File: .github/workflows/test.yml (hunk 3 above)
Comment (hamzashoukat94) [reply .github/workflows/test.yml]: This job is also renamed to `publish-code-coverage-report`

**Comment by hamzashoukat94** [comment]
We don't need to publish the code coverage for this repo because it doesn't contain any test case.

**Pull request closed without merging**